pkg/spdk/controllerserver.go

```diff
@@ -194,59 +198,69 @@ func (cs *controllerServer) DeleteSnapshot(_ context.Context, req *csi.DeleteSna
 	return &csi.DeleteSnapshotResponse{}, nil
 }
 
-func prepareCreateVolumeReq(req *csi.CreateVolumeRequest, sizeMiB int64) (*util.CreateLVolData, error) {
-	distrNdcsStr, ok := req.GetParameters()["distr_ndcs"]
-	if !ok {
-		distrNdcsStr = "1"
+func getIntParameter(params map[string]string, key string, defaultValue int) (int, error) {
+	if valueStr, exists := params[key]; exists {
+		value, err := strconv.Atoi(valueStr)
+		if err != nil {
+			return 0, fmt.Errorf("error converting %s: %w", key, err)
+		}
+		return value, nil
 	}
+	return defaultValue, nil
+}
 
-	distrNpcsStr, ok := req.GetParameters()["distr_npcs"]
-	if !ok {
-		distrNpcsStr = "1"
-	}
+func getBoolParameter(params map[string]string, key string) bool {
+	valueStr, exists := params[key]
+	return exists && (valueStr == "true" || valueStr == "True")
+}
+
+func prepareCreateVolumeReq(ctx context.Context, req *csi.CreateVolumeRequest, sizeMiB int64) (*util.CreateLVolData, error) {
+	params := req.GetParameters()
 
-	distrNdcs, err := strconv.Atoi(distrNdcsStr)
+	distrNdcs, err := getIntParameter(params, "distr_ndcs", 1)
 	if err != nil {
-		klog.Errorf("Error converting distrNdcs: %v", err)
 		return nil, err
 	}
-
-	distrNpcs, err := strconv.Atoi(distrNpcsStr)
+	distrNpcs, err := getIntParameter(params, "distr_npcs", 1)
 	if err != nil {
-		klog.Errorf("Error converting distrNpcs: %v", err)
 		return nil, err
 	}
 
-	var compression bool
-	var encryption bool
+	compression := getBoolParameter(params, "compression")
+	encryption := getBoolParameter(params, "encryption")
 
-	if req.GetParameters()["compression"] == "true" ||
-		req.GetParameters()["compression"] == "True" {
-		compression = true
-	}
+	pvcName, pvcNameSelected := params["csi.storage.k8s.io/pvc/name"]
```

Comment:
can you please create a constant: `csi.storage.k8s.io/pvc` and prepare the parameter key by appending `name` or `namespace to it.

Reply:
Sure